### PR TITLE
Adding first set up tests - very simple

### PR DESCRIPTION
**Pull request by gauntface**
These tests are just for the constructor and looking at API surface. May change when it comes to testing with Service Workers.

These current tests fail for:

  var pushClient = new window.goog.propel.Client();
  pushClient.supported();
  pushClient.hasPermission();

This is because those methods are static, you would have to do something along the lines of PushClient.supported() or in this case: window.goog.propel.Client.supported().

### CODE REVIEW

**Comment by wibblymat** [comment]
Hmm, I did actually make those static on purpose, since they don't rely on any internal state. Can certainly debate it, but I would personally change the tests rather than the API.

There is a certain amount of copy/paste from the SW Toolbox tests in evidence. I'll highlight the ones that I can spot individually, but you might want to have a read through and check that I didn't miss any.

File: test/index.html [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+  <!-- Helper functions will be under window.testHelper -->
+  <script src="/test/libs/helper-functions.js"></script>
+
+  <!-- Push Client Code We Will Test -->
+  <script src="/dist/client.js"></script>
+
+  <!--
+    Timeout is extended to ensure tests for max-cache-age
+    have enough time to complete
```
Comment (wibblymat) [inline test/index.html]: copy/pasted section

File: test/index.html [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+  <!-- In browser test scripts should be added to the page here-->
+  <script src="/test/browser-tests/push-client/push-client-tests.js"></script>
+
+  <script>
+    (function() {
+      // To automate the tests with selenium, we need a way
+      // to inform selenium when the tests are completed and
+      // pass results back.
+      // This is done by waiting for window.swtoolbox to be defined.
```
Comment (wibblymat) [inline test/index.html]: copy/pasted section

File: test/index.html [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        return {
+          title: testResult.title,
+          state: testResult.state
+        };
+      };
+
+      // Once this is called, automated tests should be
+      // able to read sw.toolbox and get test results.
+      function publishTestResults() {
+        window.swtoolbox = {
```
Comment (wibblymat) [inline test/index.html]: copy/pasted section

File: test/index.html [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+          }
+        };
+      }
+
+      // This make browsers without a service worker pass tests by
+      // bypassing the tests altogether.
+      // This is desirable to allow travis to run tests in all browsers
+      // regardless of support or not and perform tests when the browser
+      // starts to support service workers.
+      if (!('serviceWorker' in navigator)) {
```
Comment (wibblymat) [inline test/index.html]: copy/pasted section - in theory everything in Propel should properly check that support is available, so we should test in all browsers but have a modified test suite for those that don't support push. Script should still load, PushClient.supported() and PushClient.hasPermission() should still work, etc.

File: test/libs/helper-functions.js [hunk 5]
```diff
@@ -0,0 +1,185 @@
+/**
```
Comment (wibblymat) [inline test/libs/helper-functions.js]: copy/pasted file - most of this seems unused

File: test/server/index.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+// This server is needed most importantly to add the Service-Worker-Allowed
+// header to any service worker loaded in the tests. This allows the scope
+// to be manipulated any way we want / need during testing.
+
+var path = require('path');
+var express = require('express');
+var app = express();
+
+// Set up static assets
```
Comment (wibblymat) [inline test/server/index.js]: copy/pasted section

File: test/server/index.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+app.use('/test/browser-tests/',
+  express.static(path.join(__dirname, '..', 'browser-tests/'), {
+    setHeaders: function(res) {
+      res.setHeader('Service-Worker-Allowed', '/');
+    }
+  })
+);
+
+// Allow all assets in the project to be served (This includes sw-toolbox.js)
```
Comment (wibblymat) [inline test/server/index.js]: copy/pasted comment

**Comment by gauntface** [comment]
Got rid of those checks.

supported thinking about it is an edge case considering I imagine most devs will either catch the error or the static method makes sense.

hasPermission() I'm not convinced about to be honest but not important.

**Comment by gauntface** [comment]
Gone through and made commented changes.

File: test/libs/helper-functions.js (hunk 5 above)
Comment (gauntface) [inline test/libs/helper-functions.js]: removed unused methods - will add methods back in as they are needed.

**Comment by wibblymat** [comment]
:+1:
